code/modules/projectiles/guns/launcher/projectile_grenade_launcher.dm

```diff
@@ -1,4 +1,113 @@
-/obj/item/gun/projectile/shotgun/pump/china
+/obj/item/gun/projectile/shotgun/pump/grenade
+	name = "NT GL \"Protector\""
+	desc = "A bulky pump-action grenade launcher. Holds up to 6 grenades in a revolving magazine."
+	icon = 'icons/obj/guns/launcher/riotgun.dmi'
+	icon_state = "riotgun"
+	item_state = "riotgun"
+	w_class = ITEM_SIZE_BULKY
+	force = WEAPON_FORCE_PAINFUL
+	caliber = CAL_GRENADE
+	load_method = SINGLE_CASING
+	handle_casings = HOLD_CASINGS
+	max_shells = 5
+	recoil_buildup = 0
+	twohanded = TRUE
+	ammo_type = /obj/item/ammo_casing/grenade
+	fire_sound = 'sound/weapons/guns/fire/grenadelauncher_fire.ogg'
+	fire_sound_text = "a metallic thunk"
+	bulletinsert_sound = 'sound/weapons/guns/interact/batrifle_magin.ogg'	//Placeholder, could use a new sound
+	saw_off = FALSE
+	matter = list(MATERIAL_PLASTEEL = 30, MATERIAL_WOOD = 10)
+	origin_tech = list(TECH_COMBAT = 6, TECH_MATERIAL = 2)
+	price_tag = 3000
+
+/obj/item/gun/projectile/shotgun/pump/grenade/examine(mob/user)
+	if(..(user, 2))
+		if(chambered)
+			to_chat(user, "\A [chambered] is chambered.")
+
+/obj/item/gun/projectile/shotgun/pump/grenade/handle_post_fire(mob/user)
+	log_and_message_admins("fired a grenade ([chambered]) from ([src]).")
+	user.attack_log += "\[[time_stamp()]\] fired a grenade ([chambered]) from ([src])"
+	..()
+
+/obj/item/gun/projectile/shotgun/pump/grenade/proc/load_underslung(obj/item/ammo_casing/grenade/G, mob/user)
+	if(chambered)
+		to_chat(user, SPAN_WARNING("[src] is full!"))
+		return
+	user.remove_from_mob(G)
+	G.forceMove(src)
+	chambered = G
+	playsound(src.loc, bulletinsert_sound, 75, 1)
+
+/obj/item/gun/projectile/shotgun/pump/grenade/proc/unload_underslung(mob/user)
+	if(chambered)
+		var/turf/turf = get_turf(src)
+		chambered.forceMove(turf)
```

Comment:
```suggestion
		user.put_in_hands(chambered)
```
This'll put the 'nade into the user's hands if possible, otherwise it'll drop it on the ground as normal. Little bit of QOL.